src/package_validation_tool/matching/autotools.py

```diff
@@ -572,14 +572,40 @@ def _generate_autotools_files(self, tool_paths: Dict[str, Optional[str]]) -> boo
         """
         log.info("Generating Autotools files in source repository...")
 
-        # prepare PATH environment variable
+        # prepare PATH and PERL5LIB environment variables
         env = os.environ.copy()
         path_parts = [path for path in tool_paths.values() if path]
+        perl5lib_parts = []
 
         if path_parts:
             env["PATH"] = ":".join(path_parts) + ":" + env.get("PATH", "")
             log.debug("Updated PATH: %s", env["PATH"])
 
+            # Add Perl library paths for autotools modules
+            for path in path_parts:
+                if path:
+                    # Convert bin path to share directory and find Perl modules
+                    bin_path = Path(path)
+                    share_path = bin_path.parent / "share"
+                    if share_path.is_dir():
+                        # Look for directories containing .pm files
+                        for subdir in share_path.iterdir():
+                            if subdir.is_dir():
+                                # Check if this directory or its subdirectories contain .pm files
+                                pm_files = list(subdir.rglob("*.pm"))
+                                if pm_files:
+                                    perl5lib_parts.append(str(subdir))
+                                    log.debug("Found Perl modules in: %s", subdir)
```

Comment:
This cascade of nested IFs is terrible. Since we're in the `for path in path_parts` loop anyway, can we restructure to exit early (using stuff like `if not pm_files: continue`).